pkg/controller/servicebindingrequest/reconciler_test.go

```diff
@@ -277,30 +253,27 @@ func TestReconcilerGenericBinding(t *testing.T) {
 	require.Equal(t, reconcilerName, sbrOutput2.Status.Secret)
 	require.Equal(t, 1, len(sbrOutput2.Status.Applications))
 
-	// Update Credentials
+	u, err = fakeDynClient.Resource(secretsGVR).Get("db-credentials", metav1.GetOptions{})
+	require.NoError(t, err)
 	s := corev1.Secret{}
-	secretNamespaced := types.NamespacedName{Namespace: reconcilerNs, Name: "db-credentials"}
-	require.NoError(t, fakeClient.Get(ctx, secretNamespaced, &s))
-	s.Data["password"] = []byte("abc123")
-	require.NoError(t, fakeClient.Update(ctx, &s))
-
-	reconciler = &Reconciler{
-		RestMapper: restMapper,
-		client:     fakeClient,
-		dynClient:  f.FakeDynClient(),
-		scheme:     f.S,
-	}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &s)
+	require.NoError(t, err)
 
+	// Update Credentials
+	s.Data["password"] = []byte("abc123")
+	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&s)
+	require.NoError(t, err)
+	updated := unstructured.Unstructured{Object: obj}
+	_, err = fakeDynClient.Resource(secretsGVR).Namespace(updated.GetNamespace()).Update(&updated, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	time.Sleep(1 * time.Second)
```

Comment:
Is the sleep necessary here :thinking:

Reply:
Remove this will cause flaky as we inject a RFC3339 format timestamp such as "2020-05-14T08:16:42Z" to pod template. For reconciling it might fail to detect template changes since it has only second precision.